### DISCUSSION TITLE
A way to quickly zoom to each Label

### DISCUSSION THREAD

**Discussion by quitmeyer** [CVHub520/X-AnyLabeling#643] [Ideas]
Hi! This software is really really close to doing everything I need! Thanks so much! and the fact i can run it locally, and it's open source is so good.

One feature I could really use is if, after running the machine learning auto-detection, I could have some keyboard shortcuts that let you zoom the current screen to each label/detection on an image. That is, I could be looking at an image, run the model to detect where all the things are (in my case it is insects on a white background), see the full image with all the detections, but then be able to press a keyboard button that let's me look at a close view of the next label in this image. In this way I could quickly go through an image and make sure all the detections look good. Right now i spend most of my time having to zoom in, and zoom out on every single detection in an image. 

example image

![image](https://github.com/user-attachments/assets/f3d6df50-accd-49c3-ad71-806fef9af0c5)

I have to zoom in to each detection to check it's not a false positive:
![image](https://github.com/user-attachments/assets/9d570e09-6aae-4c98-a1ae-c54577b2401e)

it would be amazing if I could just switch through each label in an image at an appropriate zoom.

Maybe this feature already exists? If so i would love to learn more!

**Comment by CVHub520** [comment]
Thank you for sharing this interesting feature request. The ability to quickly zoom and navigate between detected shapes using keyboard shortcuts would indeed be very useful for efficiently reviewing and validating detections.

While we don't currently have the resources to implement this feature, we appreciate you bringing it to our attention. We're always looking for ways to improve the user experience, and your suggestion aligns well with that goal.

We encourage community contributions, so if any developers are interested in working on this feature, we'd be happy to provide guidance. In the meantime, we'll keep this idea in mind for future development priorities.

Please continue to share your thoughts and experiences using the software. User feedback is invaluable in helping us enhance the tool's functionality.

**Reply by quitmeyer** [reply]
Cool! Thanks for your reply. I do some python coding, but I'm not experienced with compiling software, so I don't know how much help i would be. I would put up a $100 bounty on this feature though if you are able to implement it. It would help out our science research project we are doing a lot!

**Reply by quitmeyer** [reply]
BTW My teammate on our project said they are happy to add another $150 to the bounty for this feature if it happens to get implemented within a week! (because it will save them a bunch of time on their current project!). No pressure of course, but if that helps a feature like this get implemented quicker we are happy to support!

**Comment by quitmeyer** [comment]
Amazing! @julian stirling implemented my request!

**Comment by quitmeyer** [comment]
Saw that you merged @julianstirling 's code! Yay! I have already been using it to churn through thousands of labels to double check them!

**Comment by CVHub520** [comment]
Great to hear you've been using this feature! I've just fixed a minor issue, so I recommend updating to the latest version for an improved experience. Thanks for your support and feedback!
